src/diffusers/models/transformers/transformer_flux.py

```diff
@@ -508,7 +508,11 @@ def custom_forward(*inputs):
             if controlnet_block_samples is not None:
                 interval_control = len(self.transformer_blocks) / len(controlnet_block_samples)
                 interval_control = int(np.ceil(interval_control))
-                hidden_states = hidden_states + controlnet_block_samples[index_block // interval_control]
+                # For Xlabs ControlNet.
+                if len(controlnet_block_samples) == 2:
```

Comment:
I would suggest passing a flag down here, `controlnet_repeat_interleave = False` maybe? 
This would break if someone trained a controlnet with 2 blocks but want to use the other indexing method